code/datums/materials/meat.dm

```diff
@@ -17,9 +17,8 @@
 
 /datum/material/meat/on_removed(atom/source, amount, material_flags)
 	. = ..()
-	if(material_flags & MATERIAL_NO_EFFECTS)
-		return
-	qdel(source.GetComponent(/datum/component/edible))
+	if(material_flags & MATERIAL_EFFECTS)
```

Comment:
move this to where this proc is called, as it should never run if theres no mat effects